### ISSUE TITLE
Accessibility change request, replace 'outline: 0 !important;' with 'box-shadow: $component-focus-box-shadow;'

### ISSUE THREAD

**Issue by georgel-pop-lr** [liferay/clay#4124]
-   [Yes ] I have searched the [issues](https://github.com/liferay/clay/issues) of this repository and believe that this is not a duplicate.

## Is there an example you can provide via codesandbox.com?
I can't provide an example via codesandbox.com

## What are the steps to reproduce?
In order to reproduce, you need to run this accessibility tool OAW described on https://issues.liferay.com/browse/LPS-133314 

In the report is complaining about the outline rule: 'outline: 0 !important;' because it thinks the keyboard focus indicator is not visible and this style makes it less accessible, it might be a false positive as I could not find an example using this style, but after talking with @pat270 he said here https://issues.liferay.com/browse/LPS-133314?focusedCommentId=2459136&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2459136 that we could change this rule with:
```
[tabindex="-1"]:focus:not(:focus-visible)
{ box-shadow: $component-focus-box-shadow; }

```
This style comes from the Clay library in https://github.com/liferay/clay/blob/e4927006d498d08936a81f75c2be0781cd9fa1fb/packages/clay-css/src/scss/bootstrap/_reboot.scss#L67, more exact in the bootstrap files.

## What is the expected result?
Change this rule  'outline: 0 !important;' and have a clean accessibility report for our client when using OAW  tool which is owned by the Government of Spain and is used by any public institution to report and verify the validity of their site requested by law.  

## Environment
master

**Comment by pat270** [comment]
This issue is also present in [3.x](https://github.com/liferay/clay/blob/168abf4255a6c0416323c23817fa04eea3d78b50/packages/clay-css/src/scss/components/_reboot.scss#L162-L164). We will need to account for https://github.com/liferay/clay/issues/3436 when we send a fix for it.

**Comment by georgel-pop-lr** [comment]
@pat270 I will need to send a fix fox 3.x? 
I understand that would be to add box-shadow on https://github.com/liferay/clay/blob/168abf4255a6c0416323c23817fa04eea3d78b50/packages/clay-css/src/scss/components/_reboot.scss#L162-L164 as of this:
`[tabindex^='-'] {	
outline: 0;
box-shadow: $component-focus-box-shadow;
}`

**Comment by pat270** [comment]
@georgel-pop-lr It's up to you. I was going to send something in today before we release tomorrow. It would be great if you have time / want to work on it.

For master, we can delete that rule set from `components/_reboot.scss` and set `outline: 0;` in modals and for `a.disabled` so we don't cause a regression for #3436.

**Comment by georgel-pop-lr** [comment]
@pat270 Sadly today I will not be able to do it, as in 15 min I will finish my work for today. So maybe is better if you can do it today before release. 

I see on modals we already have set `outline: 0; `
https://github.com/liferay/clay/blob/6fd3e9309b514a2b95f775a9551fe8446de068c7/packages/clay-css/src/scss/components/_modals.scss#L14

Regards.
